packages/dds/tree/src/feature-libraries/modular-schema/modularChangeFamily.ts

```diff
@@ -146,8 +146,14 @@ export class ModularChangeFamily
 		const genId: IdAllocator = () => brand(++maxId);
 		const crossFieldTable = newCrossFieldTable<ComposeData>();
 
+		const changesWithoutConstraintViolations = changes.filter(
+			(change) => change.change.constraintViolationCount ?? 0 === 0,
```

Comment:
I think this might not do what you expect. The way this is being parsed is `change.change.constraintViolationCount ?? (0 === 0)`, in other words `change.change.constraintViolationCount ?? true`.

This only matters in the case where `change.change.constraintViolationCount` is falsey but not `undefined` or `null` -- for numbers, 0 and NaN.

You can fix this by putting parens around the `??` operands, 
```suggestion
			(change) => (change.change.constraintViolationCount ?? 0) === 0,
```

Reply:
Good catch! My tests were failing but this might've taken me a while to track down why 😄